### ISSUE TITLE
[Dependency Injection] PHPDumper Problems in PHP8 on Linux

### ISSUE THREAD

**Issue by nanasess** [symfony/symfony#39528]
**Symfony version(s) affected**: 4.4.15

**Description**  

On Linux, the code generated by PHPDumper looks like the following and causes a Fatal Error in PHP8.

```php
<?php

include_once\dirname(__DIR__, 4).'/vendor/doctrine/persistence/lib/Doctrine/Persistence/ObjectManager.php';
include_once \dirname(__DIR__, 4).'/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManagerInterface.php';
include_once \dirname(__DIR__, 4).'/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php';
class EntityManager_9a5be93 extends \Doctrine\ORM\EntityManager implements \ProxyManager\Proxy\VirtualProxyInterface
{
```

```
!!  PHP Fatal error:  Uncaught Error: Call to undefined function include_once\dirname() in /path/to/var/cache/prod/ContainerIHdxD6S/EntityManager_9a5be93.php:3
```

On a macOS, the following code is generated and works fine.

```php
<?php
include_once \dirname(__DIR__, 4).'/vendor/doctrine/persistence/lib/Doctrine/Persistence/ObjectManager.php';
include_once \dirname(__DIR__, 4).'/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManagerInterface.php';
include_once \dirname(__DIR__, 4).'/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php';

class EntityManager_9a5be93 extends \Doctrine\ORM\EntityManager implements \ProxyManager\Proxy\VirtualProxyInterface
{

```
**How to reproduce**  

Run `cache:clear --no-warmup` on Ubuntu 18.04.

**Possible Solution**  

Generates the same code as on a macOS.

**Additional context**

**Comment by nanasess** [comment]
We have confirmed that the following ad-hoc fixes resolve the issue.

```diff
diff --git a/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php b/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php
index c4c0ddcdd224..447d26d642f5 100644
--- a/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php
+++ b/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php
@@ -560,7 +560,7 @@ private function generateProxyClasses(): array
                     if ($this->inlineFactories) {
                         $this->inlinedRequires[$file] = true;
                     }
-                    $code .= sprintf("include_once %s;\n", $file);
+                    $code .= sprintf("include_once (%s);\n", $file);
                 }
 
                 $proxyCode = $code.$proxyCode;
```
https://github.com/nanasess/symfony/commit/20b35dd1da4d51acbc24a11b38dfa2493fe3a731

**Comment by jderusse** [comment]
This is weird, the initial issue is about a missing space between `include_once` and `\dirname`, but your proposed patch, only changes brackets.
Could you confirm that, if you revert your patch (back to an initial symfony value), clear cache and rebuild the container, you reproduce the bug? In that case could you provide us a reproducer?

**Comment by nicolas-grekas** [comment]
I also fail to understand where that can come from. The line you patch does have a space after `include_once` - it's not supposed to be removed and I don't see where it can be removed.
Can you please provide us with a reproducer, or maybe debug this?

**Comment by nicolas-grekas** [comment]
I'm seeing that the code excerpt you gave doesn't have a namespace. This makes me think you might use the DI component in a not-common way. A reproducer would definitely help.

**Comment by nanasess** [comment]
@jderusse, @nicolas-grekas Thanks for your reply, I try to create reproducer.

**Comment by OskarStark** [comment]
I saw exactly such error with a newer PHP-CS-Fixer version, can you ensure it's not changed by a cs style fixer? Thanks

Note, that this is just a shot for the moon...

**Comment by nicolas-grekas** [comment]
Fixed by #39220
